src/main/java/com/mraof/minestuck/entity/animation/PhasedMobAnimation.java

```diff
@@ -64,26 +85,36 @@ public MobAnimation getAnimation()
 	/**
 	 * Equivalent to getting recoveryEnd value
 	 */
-	public int getTotalAnimationLength()
+	public int getTotalAnimationLength(double speed)
 	{
-		return recoveryEnd;
+		return (int) Math.round(recoveryEnd / speed);
 	}
 	
 	/**
 	 * Is called every tick to check whether its time to transition to a new phase
 	 */
 	public <T extends PathfinderMob & PhasedMobAnimation.Phases.Holder> void attemptPhaseChange(int time, T entity)
 	{
-		if(time == getInitiationStartTime())
+		double speed = getAnimationSpeed(entity);
+		if(time == getInitiationStartTime(speed))
 			entity.setAnimationPhase(Phases.INITIATION, animation.action());
-		else if(time == getContactStartTime())
+		else if(time == getContactStartTime(speed))
 			entity.setAnimationPhase(Phases.CONTACT, animation.action());
-		else if(time == getRecoveryStartTime())
+		else if(time == getRecoveryStartTime(speed))
 			entity.setAnimationPhase(Phases.RECOVERY, animation.action());
-		else if(time >= getTotalAnimationLength())
+		else if(time >= getTotalAnimationLength(speed))
 			entity.setAnimationPhase(Phases.NEUTRAL, animation.action());
 	}
 	
+	public double getAnimationSpeed(PathfinderMob entity)
+	{
+		if(speedModifyingAttribute == null)
+			return baseAnimationSpeed;
+		
+		AttributeInstance attributeInstance = entity.getAttribute(speedModifyingAttribute);
+		return (attributeInstance != null ? attributeInstance.getValue() : 1) * baseAnimationSpeed;
+	}
```

Comment:
It looks like you've tried tuning when the speed may change in the animation state handlers, with walk animations updating continuously while attack animations only change while an attack is not ongoing. But does that actually get reflected for stuff controlled by `PhasedMobAnimation`? What happens in here if the speed changes mid-animation?

Reply:
Good catch, I went ahead and changed it so its speed now only gets updated at the start of the AI goal